### PR TITLE
Migration to Vite

### PR DESCRIPTION
**Pull request by annybs**
This PR follows https://github.com/edge/explorer/pull/209 in an effort to adopt Vite as our primary toolchain for working with and building Vue projects.

At this time, there is at least one severe blocker that prevents merging this work: `crypto` and `buffer` are imported from Node and not polyfilled correctly. This results in the error `Uncaught (in promise) TypeError: A.sent() is null` being emitted in production builds (the development build works OK).

I have experimented with several solutions but none seem to resolve the issue. Further reading [here](https://github.com/vitejs/vite/discussions/2785) and [here](https://github.com/vitejs/vite/discussions/3126).

This work is moved to low priority until I have more time to unpick the issue.

### CODE REVIEW

**Comment by annybs** [comment]
I have further investigated the issue with bundling [crypto](https://nodejs.org/dist/latest-v18.x/docs/api/crypto.html) today, primarily with a view to removing it along with [buffer](https://nodejs.org/dist/latest-v18.x/docs/api/buffer.html), as these are large dependencies taken from Node that do not exist in the browser and are really better served by [crypto-js](https://www.npmjs.com/package/crypto-js) proper.

I have been able to successfully migrate some wallet functions, namely: create, restore, lock, and unlock the wallet. Unfortunately, export private key, send transaction etc. are not working correctly. The difference between these functions is that the former group work just by hashing, comparing, and encrypting data; _decrypting_ it (namely, the private key) is our trouble. Node's `createCipheriv()` function does not require an internal salt,

https://github.com/edge/wallet/blob/1da854ee4e05092353293ec6b956c51447a22d87/src/utils/crypto.js#L33-L47

But `CryptoJS.AES.decrypt` _does_ require the internal salt that was generated while encrypting the string input. The [documentation example](https://www.npmjs.com/package/crypto-js#API) of AES demonstrate encrypting and then decrypting a given input through the use of a single self-contained string transport. Our current implementation using Node crypto, on the other hand, uses an object transport containing hex-encoded `{ content, iv }` only (analogous to `{ ciphertext, iv }` of [CipherParams](https://github.com/brix/crypto-js/blob/4dcaa7afd08f48cd285463b8f9499cdb242605fa/src/cipher-core.js#L501-L556)) which is saved to local storage as `p2`. Because this does not include `{ salt }` the crypto-js implementation is unable to reliably decrypt the input and produces an empty string instead.

In a nutshell, without a workaround for missing salts, it is impossible for us to migrate the usage of Node crypto over to crypto-js while retaining backward compatibility. If we drop backward compatibility, this could cause loss of private keys for any users who use the browser exclusively and haven't backed it up elsewhere.

Without being able to bundle Node crypto and buffer (or at least I haven't had any luck doing so yet) this is effectively a **total blocker** for migrating from vue-cli to Vite.
